vllm/model_executor/layers/linear.py

```diff
@@ -471,6 +493,81 @@ def weight_loader(self,
         assert param_data.shape == loaded_weight.shape
         param_data.copy_(loaded_weight)
 
+    def _default_loading(self, param: vLLMParameter, param_data: torch.Tensor,
+                         loaded_weight: torch.Tensor, loaded_shard_id: int):
+        tp_rank = get_tensor_model_parallel_rank()
+        tp_size = get_tensor_model_parallel_world_size()
+
+        shard_offset = sum(self.output_sizes[:loaded_shard_id]) // tp_size
+        shard_size = self.output_sizes[loaded_shard_id] // tp_size
+
+        # TODO: Define this relationship more clearly
+        if param.is_packed and param.packed_dim == param.output_dim:
+            shard_size, shard_offset = param.adjust_packed_shard(
+                shard_offset=shard_offset, shard_size=shard_size)
+
+        param_data = param_data.narrow(param.output_dim, shard_offset,
+                                       shard_size)
+        loaded_weight.narrow(param.output_dim, tp_rank * shard_size,
+                             shard_size)
+
+        return param_data, loaded_weight
+
+    def _load_no_shard_id(self, param: vLLMParameter, loaded_weight):
```

Comment:
Let's be more explicit about what this case is. We get `no_shard_id` when the MLP and QKV are fused on disk. Let's also include a link to an example model.

- Example model: https://huggingface.co/microsoft/Phi-3-mini-4k-instruct